vspace/src/main/java/edu/asu/diging/vspace/web/staff/SpaceOverviewController.java

```diff
@@ -0,0 +1,236 @@
+
+        Iterable<SpaceLinkDisplay> spaceToSpacelinksList = spaceLinkDisplayRepository.findAll();
+
+        List<ISpaceLinkDisplay> allSpaceToSpacelinksList = new ArrayList<>();
+
+        spaceToSpacelinksList.forEach(link -> allSpaceToSpacelinksList.add(link));
+
+        Map<String, List<ISpaceLinkDisplay>> allSpaceToSpacelinksListMap = new HashMap<>();
+
+        getSpaceToAllOtherlink(allSpaceToSpacelinksList, allSpaceToSpacelinksListMap, false);
+
+        getSpaceLinkMap(request, spaceList, mapper, nodeArray, allNodeList, spaceLinkMap, allSpaceToModulelinksListMap,
+                allSpaceToSpacelinksListMap);
+
+        if (moduleList != null && !moduleList.isEmpty()) {
+            nodeFormation(request, moduleList, mapper, nodeArray, allNodeList, true);
+        }
+        sourceToTargetFormation(mapper, linkArray, allNodeList, spaceLinkMap);
+        model.addAttribute("overviewNode", mapper.writeValueAsString(nodeArray));
+        model.addAttribute("overviewLink", mapper.writeValueAsString(linkArray));
+        return "staff/spaces/graph";
+    }
+
+    /**
+     * To retrieve all space to space and space to module links corresponding to
+     * each space
+     * 
+     * @param request
+     * @param spaceList
+     * @param mapper
+     * @param nodeArray
+     * @param allNodeList
+     * @param spaceLinkMap
+     * @param allSpaceToModulelinksListMap
+     * @param allSpaceToSpacelinksListMap
+     */
+    private void getSpaceLinkMap(HttpServletRequest request, List<Space> spaceList, ObjectMapper mapper,
+            ArrayNode nodeArray, List<String> allNodeList, Map<String, List<String>> spaceLinkMap,
+            Map<String, List<IModuleLinkDisplay>> allSpaceToModulelinksListMap,
+            Map<String, List<ISpaceLinkDisplay>> allSpaceToSpacelinksListMap) {
+        if (spaceList != null && !spaceList.isEmpty()) {
+            nodeFormation(request, spaceList, mapper, nodeArray, allNodeList, false);
+            for (Space space : spaceList) {
+                List<IModuleLinkDisplay> spaceToModulelinksList = allSpaceToModulelinksListMap.get(space.getId());
+                List<ISpaceLinkDisplay> spaceLinks = null;
+                spaceLinks = allSpaceToSpacelinksListMap.get(space.getId());
+                List<ISpaceLinkDisplay> spaceToSpaceLinksList = null;
+                if (spaceLinks != null) {
+                    spaceToSpaceLinksList = new ArrayList<>();
+                    spaceToSpaceLinksList.addAll(spaceLinks);
+                }
+                List<String> spaceAllLink = new ArrayList<>();
+                if (spaceToModulelinksList != null && !spaceToModulelinksList.isEmpty()) {
+
+                    spaceToModulelinksList.forEach(link -> {
+                        spaceAllLink.add(link.getLink().getModule().getId());
+                    });
+                }
+                if (spaceToSpaceLinksList != null && !spaceToSpaceLinksList.isEmpty()) {
```

Comment:
once `spaceToSpaceLinksList` has been initialized as suggested above, you can loose this if statement;